libraries/bezier-rs/src/subpath/transform.rs

```diff
@@ -272,6 +274,183 @@ impl<ManipulatorGroupId: crate::Identifier> Subpath<ManipulatorGroupId> {
 			manipulator_group.apply_transform(affine_transform);
 		}
 	}
+
+	// Smooths a Subpath up to the first derivative, using a weighted averaged based on segment length.
```

Comment:
Doc comment with three slashes.

Reply:
I'd prefer to leave `clip_simple_subpaths` private, as it relies on two simplifying assumptions:
- The Subpaths being clipped represent only simple Bezier
- Loopbacks, the edge case I have described in the doc comments, are _very_ rare
These assumptions are both generally true while clipping the subpaths generated in `offset`/`outline`, but I don't know how useful they actually be to hypothetical users.

We could probably make `smooth_cubic_paths` public in the future, but we'd need to think about how to handle the visualization further because our visualizations allow users to play with the handles, but this function also moves handles in the smoothing process. This function also relies on some simplifying assumptions that are only true in the case of offset / outline, so we'd need to do some more work before presenting it to the users, and I don't want to bloat this PR further.

Reply:
Sounds good, that's a more than reasonable explanation, no need to make them public if it's not intended to be.